File: datasets/cpp-keep-code_davinci_001_temp_0.2/HumanEval_26_remove_duplicates.cpp
#include<assert.h>
#include<bits/stdc++.h>
// From a list of integers, remove all elements that occur more than once.
// Keep order of elements left the same as in the input.
// >>> remove_duplicates([1, 2, 3, 2, 4])
// [1, 3, 4]
std::vector<long> remove_duplicates(std::vector<long> numbers) {
    std::vector<long> result;
    std::unordered_set<long> set;
    for (long number : numbers) {
        if (set.find(number) == set.end()) {
            set.insert(number);
            result.push_back(number);
        }
    }
    return result;

}
int main() {
    auto candidate = remove_duplicates;
    assert(candidate((std::vector<long>())) == (std::vector<long>()));
    assert(candidate((std::vector<long>({(long)1, (long)2, (long)3, (long)4}))) == (std::vector<long>({(long)1, (long)2, (long)3, (long)4})));
    assert(candidate((std::vector<long>({(long)1, (long)2, (long)3, (long)2, (long)4, (long)3, (long)5}))) == (std::vector<long>({(long)1, (long)4, (long)5})));
}

File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_26_remove_duplicates.lua
-- From a list of integers, remove all elements that occur more than once.
-- Keep order of elements left the same as in the input.
-- >>> remove_duplicates([1, 2, 3, 2, 4])
-- [1, 3, 4]
local function remove_duplicates(numbers)
  local result = {}
  for _, number in ipairs(numbers) do
    table.insert(result, number)
  end
  table.sort(result)
  local last_number = result[1]
  for _, number in ipairs(result) do
    if number ~= last_number then
      table.insert(result, number)
    end
    last_number = number
  end
  return result
end

-- From a list of integers, remove all elements that occur less than once.
-- Keep order of elements left the same as in the input.
-- >>> remove_duplicates([1, 2, 3, 2, 4])
-- [1, 2]

lu = require('luaunit')

function test_humaneval()
local candidate = remove_duplicates
    lu.assertEquals(candidate({}), {})
    lu.assertEquals(candidate({1, 2, 3, 4}), {1, 2, 3, 4})
    lu.assertEquals(candidate({1, 2, 3, 2, 4, 3, 5}), {1, 4, 5})
end

os.exit(lu.LuaUnit.run())